modules/edu.gemini.seqexec.engine/src/main/scala/edu/gemini/seqexec/engine/model.scala

```diff
@@ -10,7 +10,10 @@ import Scalaz._
 case class QState(pending: Queue[Action],
                   current: Current,
                   done: Queue[Result],
-                  status: Status)
+                  status: Status) {
+
+  def isEmpty: Boolean = this.pending.sequences.isEmpty && this.current.isEmpty
```

Comment:
I was reluctant to use `this` but wanted to make `isEmpty` a method of `Queue` instead of a standalone function over `Queue` because it seems more natural for a publicly exposed API.

Is there a more functional way to implement this kind of methods? The Syntax Ops packages seem a bit overkill but perhaps they are worth writing for these cases anyway.